app/lib/actions/katello/content_view_version/incremental_update.rb

```diff
@@ -89,10 +89,12 @@ def plan(old_version, environments, options = {})
               end
 
               sequence do
-                new_puppet_environment = plan_action(Katello::ContentViewPuppetEnvironment::Clone, old_version,
-                                                   :new_version => new_content_view_version).new_puppet_environment
-                check_puppet_module_duplicates(content[:puppet_module_ids])
-                copy_action_outputs += copy_puppet_content(new_puppet_environment, content[:puppet_module_ids], old_version) unless content[:puppet_module_ids].blank?
+                if SmartProxy.pulp_primary.has_feature?(SmartProxy::PULP_FEATURE) && SETTINGS[:katello][:content_types][:puppet]
```

Comment:
Not too familiar with the code, but does it make sense to have the `if` outside of `sequence`?

Reply:
ah yeah, good point. I'll move it.

Reply:
it really doesn't matter in this case.  The following is basically the same:

```
if blah
  sequence 
  do 
end
```
```
sequence do 
  if blah
  end
end
```

The first one would technically be slightly more efficient, but really it shouldn't matter

Reply:
I figured it would indeed be a micro-optimization, depending on how expensive a call to `sequence` is.